### ISSUE TITLE
Stremio using too much RAM and freezing Ubuntu

### ISSUE THREAD

**Issue by AXL1994** [Stremio/stremio-bugs#308]
**Describe the bug**
I'm using Stremio (Flatpak version) on Ubuntu 21.04. When I watch a series, Stremio start the next episode when I finish the previous one. When I watch the first episode everything is ok, Stremio and Ubuntu are very smooth, but when the next episode start, everything start to freeze and lag.

The RAM is saturated, like if Stremio is keeping in memory the previous episode. It makes the computer unusable, and I have to force shutdown because it's so unresponsive that I can't click the shutdown button.

**To Reproduce**
- download the flatpak version to Ubuntu
- start an episode
- when it's at 100% start the next one

You will see that the RAM keeps being filled, until the computer is unusable.

**Expected behavior**
The movie/series should be cached on the hard disk, instead they are cached on RAM without any limit

**Screenshots**
See attachments

**Desktop (please complete the following information):**
 - OS: Ubuntu 21.04
 - 4 Gb of RAM, 8 Gb of Swap
 

**RAM usage with Stremio running**
![RAM with Stremio](https://user-images.githubusercontent.com/49454228/126410834-1f3dd3ee-bd43-4a8e-a061-c12263bcca0a.png)

**RAM usage details**
![RAM details](https://user-images.githubusercontent.com/49454228/126410838-59d991e0-66a8-4a4d-8b0a-7dcaae7bee0b.png)

**RAM used after closing Stremio**
![RAM without Stremio](https://user-images.githubusercontent.com/49454228/126410837-eb104d78-4f26-4a69-b8d1-2eb19e8148d8.png)

**Comment by TRtomasz** [comment]
Hi,
Could you please tell us how did you install stremio? Was it with flatpak package?

**Comment by AXL1994** [comment]
> 
> 
> Hi,
> Could you please tell us how did you install stremio? Was it with flatpak package?

Yes, it was a flatpak package. In the past I tried the .deb file from the site, but I had some problems installing it, so at the end I used the flatpak package, it's much easier to install. So I can't tell if the problem is present also in the .deb version, in the flatpak is present for sure. I use Ubuntu with only some Gnome extentions and nothing more, I use it mostly as a media player.

**Comment by AXL1994** [comment]
@TRtomasz 
In the meantime I changed distro, I installed Fedora 34 and installed the Flatpak package as well, but the problem is still here. 
I searched on the web about memory leaks with node.js, because it fills up a lot of RAM, and I found this link where they are talking about a memory leak with PM2 on: https://github.com/nodejs/node/issues/38120

I don't know if it's useful, furthermore the flatpak package is outdated at the version 4.4.137, so maybe in the 4.4.142 version the bug is fixed.

**Comment by TRtomasz** [comment]
@AXL1994  thanks for the update. @core1024  could we do anything with this? I'm pretty sure that leak is somehow caused by flatpak.

**Comment by kityr** [comment]
Am experiencing this on Fedora 37 Silverblue with Flathub version of Stremio. Ram just fills up, my hunch is that the media watched stays in RAM for some reason.

**Comment by hyuri** [comment]
Same here. Flatpak version. Used to happen almost every time I used Stremio: Fills memory up, freezes whole desktop.

It almost never happens now that I set up zram, which more than doubled my available memory, but the slow downs still happen and the desktop freeze did happen once since I set set up zram.

Pop!_OS 21.10
Gnome 40

**Comment by rafheros** [comment]
I think this is the root cause also of the app causing restarts on LG WebOS TV's and Firestick devices. I think the app is no cleaning the buffer of watched media  until the media is done, this is a problem when almost all Full HD content today has more then 1GB of size, smart tv devices has low memory and this turn the app unusable in these devices

**Comment by TRtomasz** [comment]
Closing this issue as we have released new app version https://blog.stremio.com/stremio-5-released-for-linux/
